### PR TITLE
Add parity test for simple RNN

### PR DESCRIPTION
**Pull request by mpariente**
This is a simple test for a shifted and summed random dataset. 
The test passes on CPU (I added the GPU restriction before pushing). 

## What does this PR do?
Compares basic lightning training to vanilla torch training for RNN. 

#### Note 
Even for the architecture for which we have performance difference between PL 0.6.0 and PL0.7.1 (as discussed in #1136), the test passes, for both versions. 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by williamFalcon** [comment]
@mpariente so haha... where does that leave us? sounds like these parity tests show lightning is working as expected? is it maybe truncated backprop? maybe we can add it to this if it already isn't?

**Comment by mpariente** [comment]
I was hoping for a little difference but even to `1e-12`, it was still passing.. It's great but it's so frustrating ! You saw the tensorboard curves in #1136 right? I'm sure there's something, our code is the same, environments the same.. 

Anyway, this parity is just the tip of the iceberg, we don't test for any real  lightning features. But this is a good start. 

Has anything major changed in the callbacks behavior, or schedulers between 0.6.0 and 0.7.1?

**Comment by williamFalcon** [comment]
@mpariente yeah, those curves hint at learning rate mismatch or learning rate scheduler mismatch. maybe you guys changed the scheduler? or maybe we do something different for scheduler?

i think there was something about .step vs .epoch for the scheduler.
@jeremyjordan 

something like this:
https://github.com/PyTorchLightning/pytorch-lightning/issues/1333

**Comment by Borda** [comment]
we forget to add it to changelog...
